### ISSUE TITLE
ksh93: read -r doesn't work if -d is also specified

### ISSUE THREAD

**Issue by zakukai** [att/ast#37]
(This is using ksh 93u+ 2012-08-01 on RHEL7)

Hi, I'm really hoping Korn Shell development will continue. Right now it kind of looks like a dead project - which is a shame because it's probably still the best of the Unix shells.

Anyway, I'm writing this shell library called "shell-pepper" and in the course of thinking about how to write a version of "read" that would read a single JSON value from the input (and stop at the end - and without writing it as a loadable "built-in") it led to this line of experimentation with the built-in "read":

```
$ a='foo\ bar }'
$ echo "'$a'"      # sanity check that $a contains what I expect
'foo\ bar }'
$ IFS='' read -r -d '}' x <<<"$a"     # Read to the next '}' or EOF.  $? tells us whether it was a delimiter or EOF.
$ echo "'$x'"                       # Thanks to IFS the space at the end is preserved, but we lose the backslash.
'foo bar '
$ IFS='' read -r y <<<"$a"    # If I remove -d, then the backslash is retained, but I lose the ability to stop the read at the next curly brace
$ echo "'$y'"
'foo\ bar }'
```

Basically the idea here is that if I've started reading a JSON object, reading to the next '}' may not get me _to_ the end of the object, but it certainly won't take me _past_ the end of the object. But I need backslashes intact (hence the -r), but when I use "-d" as well, backslashes in the input are lost (as if -r weren't specified)

BASH gets this one right:
```
a='foo\ bar }'
$ echo "'$a'"
'foo\ bar }'
$ IFS='' read -r -d '}' x <<<"$a"
$ echo "'$x'"
'foo\ bar '
```

_As a side note - I had heard that JSON read and write were to be added to Korn Shell in upcoming versions. I am mostly using 93u but with a couple 93v builds kicking around. At the time I wrote this I was unaware that the JSON functionality is already present in 93v. Nice!_

**Comment by zakukai** [comment]
...Actually, it appears this bug is already fixed in 93v (Version ABIJM 93v- 2014-12-24), and there's a test covering the issue...  So I guess this one should be closed.
